Make `ember-application` package tests pass without jQuery.

rwjblue opened the pull request:
This gets the `ember-application` package test suite passing when jQuery is not loaded.

To test a given package without jQuery, navigate to:

```
http://localhost:4200/tests/index.html?hidepassed&jquery=none&package=ember-application
```

Primary changes:

* Begin removing jQuery usage in application tests.
* Create shared qunit-fixture setup functionality.
* Remove jQuery usage in abstract test cases.
* Implement native DOM event dispatcher.
* Remove eager jQuery destructuring.

Commit:
Remove eager jQuery destructuring.

Commit:
Implement native DOM event dispatcher.

Leverages work done by ember-native-dom-event-dispatcher addon, and
switches implementation based on jQuery presence.

Commit:
Remove jQuery usage in ember-application tests.

Commit:
Remove jQuery usage in abstract test cases.

Commit:
Create shared qunit-fixture setup functionality.

rwjblue commented on packages/ember-application/tests/system/application_test.js:
```diff
@@ -36,12 +36,13 @@ import {
 moduleFor('Ember.Application, autobooting multiple apps', class extends ApplicationTestCase {
   constructor() {
-    jQuery('#qunit-fixture').html(`
+    let fixture = document.querySelector('#qunit-fixture');
```
Update this to use `get fixture() {}`

chancancode commented on packages/ember-application/tests/system/bootstrap-test.js:
```diff
@@ -25,6 +23,6 @@ moduleFor('Ember.Application with default resolver and autoboot', class extends
   ['@test templates in script tags are extracted at application creation'](assert) {
     this.runTask(() => this.createApplication());
-    assert.equal(this.$('#app').text(), 'Hello World!');
+    assert.equal(document.querySelector('#app').textContent, 'Hello World!');
```
This can be `getElementById`, but we can/should probably just make `#app` `this.element` so you can just use `this.assertText`

chancancode commented on packages/ember/tests/helpers/link_to_test.js:
```diff
@@ -655,7 +655,7 @@ moduleFor('The {{link-to}} helper - nested routes and link-to arguments', class
         {{#each model as |person|}}
           <li>
             {{#link-to 'item' person}}
```
Why not `id=person.id` here too?

rwjblue replied:
Good point, will update to that.

chancancode commented on packages/internal-test-helpers/lib/test-cases/abstract.js:
```diff
@@ -43,6 +48,11 @@ export default class AbstractTestCase {
   }
 
+  setupFixture(innerHTML) {
+    let fixture = document.querySelector('#qunit-fixture');
```
`getElementById`?

chancancode commented on packages/internal-test-helpers/lib/test-cases/abstract.js:
```diff
@@ -88,11 +98,17 @@ export default class AbstractTestCase {
 
   textValue() {
-    return this.$().text();
+    return this.element ? this.element.textContent : '';
```
does anything break without the guard?

rwjblue replied:
Yes, some of the `visit` tests with `shouldRender: false` throw without the guard. Specifically they are trying to confirm that nothing has rendered. 

I could remove the guard, and update those handful of tests to simply use `assert.equal(document.getElementById('qunit-fixture').children.length, 0)`.

chancancode replied:
I see.. maybe put that in an `assertNothingRendered` method or something. Perhaps we can just assert `this.element = null`.. but I am not sure if that semantically works for all the tests.

rwjblue replied:
It was only one test after the other visit test refactors that I did so I just put an inline assertion there...

chancancode reviewed (approved):
LGTM

Commit:
Run tests without jQuery (package by package).

Adds support for running tests per-package without jQuery so that jQuery
requirements do not creep into the rest of the framework.

wycats committed:
Fix stray `#` in change to getElementById
